app/controllers/teachers_controller.rb

```diff
@@ -80,11 +80,10 @@ def update
     @teacher.save!
     @school.save!
     if is_admin?
-      flash[:success] = "Saved #{@teacher.full_name}"
-      redirect_to teachers_path, notice: "Successfully updated information"
+      redirect_to teachers_path, notice: "Saved #{@teacher.full_name}"
     else
       # Resends emails only when teacher updates
-      TeacherMailer.form_submission(@teacher).deliver_now
+      TeacherMailer.welcome_email(@teacher).deliver_now
```

Comment:
oops. This was my bad -- I got confused by the status. I think you now have more recent refactoring that makes this more clear.